src/main/java/bio/terra/app/controller/SnapshotsApiController.java

```diff
@@ -313,4 +314,25 @@ public ResponseEntity<List<String>> retrieveUserSnapshotRoles(UUID id) {
     List<String> roles = snapshotService.retrieveUserSnapshotRoles(id, getAuthenticatedInfo());
     return new ResponseEntity<>(roles, HttpStatus.OK);
   }
+
+  @Override
+  public ResponseEntity<SnapshotLinkDuosDatasetResponse> linkDuosDatasetToSnapshot(
+      UUID id, String duosId) {
+    AuthenticatedUserRequest userReq = getAuthenticatedInfo();
+    iamService.verifyAuthorization(
+        userReq, IamResourceType.DATASNAPSHOT, id.toString(), IamAction.SHARE_POLICY_READER);
+    SnapshotLinkDuosDatasetResponse response =
+        snapshotService.updateSnapshotDuosDataset(id, userReq, duosId);
+    return new ResponseEntity<>(response, HttpStatus.OK);
```

Comment:
It's a minor thing but using `ResponseEntity.ok()` is a shorter way to write this. Also personally I would inline the method call 
```suggestion
    return ResponseEntity.ok(snapshotService.updateSnapshotDuosDataset(id, userReq, duosId));
```

Reply:
Good tip, thanks!